ValkeyJSON.md

```diff
@@ -0,0 +1,214 @@
+driven by LL(1) grammar. The core class is Selector, which glues everything together, and makes the JSONPath query apply 
+to all CRUD operations. Internally, the Selector maintains two pointers. One points to the current position in the query 
+path string. The other points to the current node in the JSON tree. Based on what’s parsed so far, it takes actions to 
+traverse the tree or select target values.
+
+The Selector is designed to work with a vector of values instead of a single value, and compatible with both RedisJSON 
+v1 and v2 query syntax. It automatically detects if the query is of v1 or v2 syntax, and sets the member isV2Path. 
+Member isWriteMode indicates READ/WRITE mode, which is automatically set based on the entry point method being invoked, 
+which is getValues, setValues or deleteValues.
+
+Insert and update are implemented as a 2-phase operation that splits the write operation into two calls - prepareSetValues 
+and commit, where prepareSetValues does not change the Valkey data. The purpose of having a 2-phase write is to be able 
+to discard the write operation if certain conditions are not satisfied.
+
+### Tokenization of JSON Object Keys
+
+JSON has two container types - Array and Object. The JSON Object is a key/value mapping where keys are restricted to 
+be strings, whereas the value can be any valid JSON value, including another container value. Thus, a single JSON document 
+can contain multiple JSON objects, each with its own unique namespace. It’s quite common that a key repeatedly appears 
+in the same JSON document or across documents. If the number of JSON documents is large, repeated key names can consume 
+significant amount of memory usage.
+
+To remove duplicate copies of JSON object key names, the module tokenizes key names by maintaining a global data structure 
+called KeyTable, which is a sharded hash table storing key tokens and reference counts. Access to the KeyTable is threadsafe.
+
+## Specification
+
+### Supported JSON Standard
+
+[RFC7159](http://www.ietf.org/rfc/rfc7159.txt) and [ECMA-404](http://www.ecma-international.org/publications/standards/Ecma-404.htm) 
+JSON data interchange standard is supported. UTF-8 Unicode in JSON text is supported.
+
+### Root JSON Value
+
+In earlier RFC 4627, only objects or arrays were allowed as root values of JSON. Since [RFC 7159](http://www.ietf.org/rfc/rfc7159.txt), 
+the root value of a JSON document can be of any type, scalar type (String, Number, Boolean, Null) or container type (Array, Object). 
+ValkeyJSON is compliant with [RFC 7159](http://www.ietf.org/rfc/rfc7159.txt).
+
+### JSON Command API
+
+There are 22 JSON commands. See https://docs.aws.amazon.com/memorydb/latest/devguide/json-list-commands.html. The API is 
+compatible with RedisJSON v2. Note that API compatibility here means our command API is a superset of RedisJSON API. 
+For example, we have command “JSON.DEBUG DEPTH” and “JSON.DEBUG FIELDS”, while they do not.
```

Comment:
Ideally I would list them all out here. I don't think it makes a lot of sense to reference the AWS documentation for the set of commands.

Reply:
It would also help to call out and give details about _all_ the commands that we add.

Reply:
Will document all commands

Reply:
Thanks for adding the command specifications.  That is most helpful.

Which commands are in the Valkey superset of the RedisJSON API?  So far as I can tell, only JSON.DEBUG (with 5 supported subcommands).  Is that all?  Is it intended that all extensions be subcommands of JSON.DEBUG?